modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenTestCase.java

```diff
@@ -0,0 +1,577 @@
+    }
+
+    private void assignRoleToUser(String roleID, String userID) throws IOException {
+        RoleItemAddGroupobj rolePatchReqObject = new RoleItemAddGroupobj();
+        rolePatchReqObject.setOp(RoleItemAddGroupobj.OpEnum.ADD);
+        rolePatchReqObject.setPath(USERS);
+        rolePatchReqObject.addValue(new ListObject().value(userID));
+        scim2RestClient.updateUserRole(new PatchOperationRequestObject().addOperations(rolePatchReqObject), roleID);
+    }
+
+    /**
+     * Authorizes system APIs and business APIs.
+     *
+     * @param applicationId Application id.
+     * @param customScopes
+     * @throws Exception
+     */
+    private void authorizeAPIs(String applicationId, List<String> customScopes, String externalServiceName, String externalServiceURI) throws Exception {
+        // Authorizes a few system APIs
+        if (!CarbonUtils.isLegacyAuthzRuntimeEnabled()) {
+            oAuth2Service.authorizeSystemAPIs(applicationId, new ArrayList<>(Arrays.asList(SCIM2_USERS_API, ACTIONS_API,
+                    APPLICATION_MANAGEMENT_API, API_RESOURCE_MANAGEMENT_API)));
+        }
+
+        // Creates business API
+        BusinessAPICreationModel businessAPICreationModel = new BusinessAPICreationModel();
+        businessAPICreationModel.setName(externalServiceName);
+        businessAPICreationModel.setIdentifier(externalServiceURI);
+        businessAPICreationModel.setDescription("This is a test external service");
+        businessAPICreationModel.setRequiresAuthorization(REQUIRES_AUTHORIZATION);
+        List<ScopeGetModel> newScopes = new ArrayList<>();
+        customScopes.forEach(scope -> {
+            ScopeGetModel newCustomScope = new ScopeGetModel();
+            newCustomScope.setName(scope);
+            newCustomScope.setDescription("This is a test scope");
+            newCustomScope.setDisplayName(scope);
+            newScopes.add(newCustomScope);
+        });
+        businessAPICreationModel.setScopes(newScopes);
+        String businessAPIID = oAuth2Service.createBusinessAPIs(businessAPICreationModel);
+
+        // Authorizes business APIs
+        AuthorizedAPICreationModel authorizedBusinessAPICreationModel = new AuthorizedAPICreationModel();
+        authorizedBusinessAPICreationModel.setId(businessAPIID);
+        authorizedBusinessAPICreationModel.setPolicyIdentifier(RBAC);
+        authorizedBusinessAPICreationModel.setScopes(customScopes);
+        oAuth2Service.authorizeBusinessAPIs(applicationId, authorizedBusinessAPICreationModel);
+    }
+
+    /**
+     * Provides data for creating pre issue access token action
+     * Each dataset consists of an endpoint uri and authentication type.
+     *
+     * @return Two-dimensional array containing pairs of endpoint uris and authentication types.
+     */
+    @DataProvider(name = "getEndpointDetails")
+    public Object[][] getEndpointDetails() {
+
+        return new Object[][]{
+                {"https://wso2is.free.beeceptor.com", "Basic"}
```

Comment:
```suggestion
                {"https://wso2is.free.beeceptor.com", AuthenticationType.BASIC}
```

Rather than passing the string we can directly pass the Enum.

Reply:
Altered the implementation with 6e4aab9c2f07c6fa7ac9e7d13b2ea705224a25a5